lib/resque/failure/redis.rb

```diff
@@ -48,14 +48,18 @@ def self.all(offset = 0, limit = 1, queue = nil)
       end
 
       def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil, order = 'desc')
+        if class_name
+          original_limit = limit
+          limit = count
```

Comment:
Why do we even get the count?  It seems like it should be possible to use a `nil` limit (i.e. `-1` limit in the redis lrange command)